src/cart/crt_hg.c

```diff
@@ -903,6 +903,76 @@ crt_hg_class_init(crt_provider_t provider, int ctx_idx, bool primary, int iface_
 	return rc;
 }
 
+static void
+crt_hg_ctx_init_tm(struct crt_hg_context *hg_ctx, int idx)
+{
+	struct crt_hg_metrics *metrics;
+	char                  *prov;
+	int                    rc = 0;
+
+	if (hg_ctx == NULL) {
+		D_ERROR("hg_ctx is NULL.\n");
+		return;
+	}
+
+	prov    = crt_provider_name_get(hg_ctx->chc_provider);
+	metrics = &hg_ctx->chc_metrics;
+
+	rc = d_tm_add_metric(&metrics->chm_bulks, D_TM_COUNTER,
+			     "Mercury-layer count of bulk transfers", "bulks",
+			     "net/%s/hg/bulks/ctx_%u", prov, idx);
```

Comment:
Would need to either handle multiprovider case here or add todo for it.
for multiprovider situation the name would need to have a form of  'net/[provider secondary idx]/[provider_name]/hg/bulks/[ctx_idx]' as the same provider can be used both as a primary and a secondary provider.

Reply:
Hmm. I guess that comment also applies to the [existing CaRT-level metrics](https://github.com/daos-stack/daos/blob/master/src/cart/crt_context.c#L280)?

Can we handle all of that in a standalone PR that is specifically for multiprovider metrics?

Reply:
@kjacque: FYI

Reply:
Indeed it would also apply to already existing cart level metrics, so yeah probably best to handle it all in a standalone pr later

Reply:
This is [DAOS-8704](https://daosio.atlassian.net/issues/DAOS-8704) btw. I will address them all with that ticket.